mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb

```diff
@@ -47,28 +47,43 @@ func resourceComputeRouterInterface() *schema.Resource {
 			},
 			"vpn_tunnel": {
 				Type:             schema.TypeString,
-				ConflictsWith:    []string{"interconnect_attachment"},
 				Optional:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: compareSelfLinkOrResourceName,
-				AtLeastOneOf:     []string{"vpn_tunnel", "interconnect_attachment", "ip_range"},
-				Description:      `The name or resource link to the VPN tunnel this interface will be linked to. Changing this forces a new interface to be created. Only one of vpn_tunnel and interconnect_attachment can be specified.`,
+				AtLeastOneOf:     []string{"ip_range", "interconnect_attachment", "subnetwork", "vpn_tunnel"},
+				ConflictsWith:    []string{"interconnect_attachment", "subnetwork"},
+				Description:      `The name or resource link to the VPN tunnel this interface will be linked to. Changing this forces a new interface to be created. Only one of vpn_tunnel, interconnect_attachment or subnetwork can be specified.`,
 			},
 			"interconnect_attachment": {
 				Type:             schema.TypeString,
-				ConflictsWith:    []string{"vpn_tunnel"},
 				Optional:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: compareSelfLinkOrResourceName,
-				AtLeastOneOf:     []string{"vpn_tunnel", "interconnect_attachment", "ip_range"},
-				Description:      `The name or resource link to the VLAN interconnect for this interface. Changing this forces a new interface to be created. Only one of vpn_tunnel and interconnect_attachment can be specified.`,
+				AtLeastOneOf:     []string{"ip_range", "interconnect_attachment", "subnetwork", "vpn_tunnel"},
+				ConflictsWith:    []string{"subnetwork", "vpn_tunnel"},
+				Description:      `The name or resource link to the VLAN interconnect for this interface. Changing this forces a new interface to be created. Only one of interconnect_attachment, subnetwork or vpn_tunnel can be specified.`,
 			},
 			"ip_range": {
+				Type:           schema.TypeString,
+				Optional:       true,
+				ForceNew:       true,
+				Computed:       true,
+				Description:    `The IP address and range of the interface. The IP range must be in the RFC3927 link-local IP space. Changing this forces a new interface to be created.`,
+			},
+			"private_ip_address": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				AtLeastOneOf: []string{"vpn_tunnel", "interconnect_attachment", "ip_range"},
```

Comment:
Can we preserve this `AtLeastOneOf`? It's theoretically covered as it still exists on other fields, but having it here as well is consistent w/ the approach throughout the rest of the codebase.

Reply:
Added, thanks!